### DISCUSSION TITLE
Project with two different layouts

### DISCUSSION THREAD

**Discussion by j314h** [remix-run/remix#4195] [General]
Hello everyone, I'm testing remix. And in my test project, I wanted to make a (public) layout for the part of the site visible to a non-login user. And a (private) layout so that logged in users can access it.

I found on the internet (not on the remix documentation), that I had to do something like:
```go
/* layout 1 in index.tsx */
routes/__index.tsx
routes/__index/index.tsx
routes/__index/login.tsx

/* layout 2 in admin.tsx */
routes/admin.tsx
routes/admin/index.tsx
routes/admin/posts/index.tsx
routes/admin/posts.tsx
```

this structure with __ allows me to put a layout in
__index.tsx
and a different layout in
admin.tsx

my question is :
- am I doing well? is the right way to do it? (most recommended)
- if I am doing things wrong can you give me a clue because I did not find any part in the docs that deals with this kind of case.

Thanks for your future response.

**Comment by phuoccss** [comment]
You are doing it correctly. The docs can be found here https://remix.run/docs/en/v1/api/conventions#file-name-conventions

**Reply by j314h** [reply]
wow thank you very much!!
